src/components/Attachments/AttachmentCarousel/Pager/index.tsx

```diff
@@ -36,13 +36,18 @@ type AttachmentCarouselPagerProps = {
 };
 
 function AttachmentCarouselPager(
-    {items, scrollEnabled = true, renderItem, initialIndex, onTap, onPageSelected, onScaleChanged}: AttachmentCarouselPagerProps,
+    {items, isZoomedOut = true, renderItem, initialIndex, onTap, onPageSelected, onScaleChanged}: AttachmentCarouselPagerProps,
     ref: ForwardedRef<AttachmentCarouselPagerHandle>,
 ) {
     const styles = useThemeStyles();
     const pagerRef = useRef<PagerView>(null);
 
-    const isPagerSwiping = useSharedValue(false);
+    const isPagerScrolling = useSharedValue(false);
+    const isScrollEnabled = useSharedValue(isZoomedOut);
+    useEffect(() => {
+        isScrollEnabled.value = isZoomedOut;
+    }, [isScrollEnabled, isZoomedOut]);
+
```

Comment:
Controlling a value from prop using `useEffect` feels a little workaround-ish (it only triggers on change). Isn't `isScrollEnabled` identical to `isZoomedOut`? 

Also, why we are setting `isScrollEnabled` to `true` on the `pageScrollHandler`. If the scroll handler got called then I'm expecting that `isScrollEnabled` is already true.

Reply:
> Controlling a value from prop using useEffect feels a little workaround-ish (it only triggers on change). Isn't isScrollEnabled identical to isZoomedOut?

The reason why we have this `isScrollEnabled` shared value, is so that we can enable/disable it from `AttachmentViewPdf/index.android.js`.

> Also, why we are setting isScrollEnabled to true on the pageScrollHandler. If the scroll handler got called then I'm expecting that isScrollEnabled is already true.

You're right. This is redundant...

Reply:
What concerns me here is that we have two values (`isZoomedOut`, `isScrollEnabled`) that refer to the same thing but could be out-of-sync. I checked in `AttachmentCarousel` and I don't think we need the `isZoomedOut` logic there, so we can just move it to `AttachmentCarouselPager`